### PR TITLE
Multi field names may not contain dots

### PR DESCRIPTION
**Pull request by brwe**
related to #14957

### CODE REVIEW

**Comment by brwe** [comment]
@rjernst can you take a look?

File: core/src/test/java/org/elasticsearch/index/mapper/multifield/MultiFieldTests.java [hunk 1]
```diff
@@ -536,4 +526,29 @@ public void testNestedFieldNotAllowed() throws Exception {
...
+                .endObject()
+                .endObject()
+                .endObject();
+
+        MapperService mapperService = createIndex("test").mapperService();
+        try {
+            mapperService.documentMapperParser().parse(mapping.string());
+            fail("this should throw an exception because one field contains a dot");
+        } catch (MapperParsingException e) {
+        }
```
Comment (rjernst) [inline core/src/test/java/org/elasticsearch/index/mapper/multifield/MultiFieldTests.java]: Add an assert that the exception was what we expect (check part of the message)?

File: core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java [hunk 2]
```diff
@@ -325,6 +325,9 @@ public static boolean parseMultiField(FieldMapper.Builder builder, String name,
 
             for (Map.Entry<String, Object> multiFieldEntry : multiFieldsPropNodes.entrySet()) {
                 String multiFieldName = multiFieldEntry.getKey();
+                if (multiFieldName.contains(".")) {
+                    throw new MapperParsingException("Field name [" + multiFieldName + "] cannot contain '.'");
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java]: Maybe this message should give more context, that this is a multifield for another field (and include that field path)?

**Comment by rjernst** [comment]
Looks good, a few suggestions on the exception and test.

**Comment by brwe** [comment]
@rjernst Thanks for reviewing so quickly! I changed according to your suggestions.

**Comment by rjernst** [comment]
Lgtm

**Pull request merged**